mnefun/_mnefun.py

```diff
@@ -2503,20 +2501,24 @@ def _head_pos_annot(p, raw_fname, prefix='  '):
     head_pos = read_head_pos(pos_fname)
 
     # do the annotations
-    lims = [p.rotation_limit, p.translation_limit, p.coil_dist_limit]
+    lims = [p.rotation_limit, p.translation_limit, p.coil_dist_limit,
+            p.coil_t_step_min, p.coil_t_window]
     if not np.isfinite(lims).any():
         annot = None
     else:
         annot_fname = raw_fname[:-4] + '-annot.fif'
         if not op.isfile(annot_fname):
             raw = mne.io.read_raw_fif(raw_fname, allow_maxshield='yes')
-            print('%sAnnotating raw segments with rotation_limit=%s, '
-                  'translation_limit=%s, coil_dist_limit=%s'
-                  % (prefix, lims[0], lims[1], lims[2]))
+            print(prefix.join(['', 'Annotating raw segments with:\n',
+                               '  rotation_limit=%s\n' % lims[0],
+                               '  translation_limit=%s\n' % lims[1],
+                               '  coil_dist_limit=%s, t_step=%s, t_window=%s'
+                               % tuple(lims[2:])]))
             annot = annotate_head_pos(
                 raw, head_pos, rotation_limit=lims[0],
                 translation_limit=lims[1],
-                coil_dist_limit=lims[2], prefix='  ' + prefix)
+                coil_dist_limit=lims[2], t_step=lims[3], t_window=lims[4],
+                prefix='  ' + prefix)
             annot.save(annot_fname)
```

Comment:
Are annotation parameters now mandatory?
> Traceback (most recent call last):
>   File ".../mnefun_/process_mismatch.py", line 157, in <module>
>     print_status=True
>   File .../mnefun/mnefun/_mnefun.py", line 627, in do_processing
>     outs[ii] = func(p, subjects, structurals, run_indices)
>   File ".../mnefun/mnefun/_report.py", line 96, in gen_html_report
>     for ri, fname in enumerate(fnames)]
>   File ".../mnefun/mnefun/_report.py", line 96, in <listcomp>
>     for ri, fname in enumerate(fnames)]
>   File ".../mnefun/mnefun/_mnefun.py", line 2522, in _head_pos_annot
>     annot.save(annot_fname)
> AttributeError: 'NoneType' object has no attribute 'save'

Reply:
No, sounds like a bug

Reply:
I think it's this bit, evaluates `False` by default
```
if not np.isfinite(lims).any():
    annot = None
```

Reply:
That part is fine, if no annotations are requested none should be saved. We should just add a `if annot is not None` before this `.save` line.

Reply:
You want me to do it?

Reply:
Nah it can be part of this PR

Reply:
Ok. So that I understand, we are annotating by default for `twa` and if limits are defined annotations are saved for downstream use?

Reply:
>  we are annotating by default for twa and if limits are defined annotations are saved for downstream use?

Not by default for TWA (or any other `trans_to` param). You need to enable annotations explicitly by setting some finite requirement values.